spec/lib/vanagon/component/source/http_spec.rb

```diff
@@ -10,21 +10,57 @@
   let (:plaintext_url) { "#{base_url}/#{plaintext_filename}" }
   let (:plaintext_dirname) { './' }
   let (:md5sum) { 'abcdssasasa' }
+  let (:sha256sum) { 'foobarbaz' }
+  let (:sha512sum) { 'teststring' }
   let (:workdir) { "/tmp" }
 
   describe "#dirname" do
     it "returns the name of the tarball, minus extension for archives" do
-      http_source = Vanagon::Component::Source::Http.new(tar_url, sum: md5sum, workdir: workdir)
+      http_source = Vanagon::Component::Source::Http.new(tar_url, sum: md5sum, workdir: workdir, sum_type: "md5")
       expect(http_source).to receive(:download).and_return(tar_filename)
       http_source.fetch
       expect(http_source.dirname).to eq(tar_dirname)
     end
 
     it "returns the current directory for non-archive files" do
-      http_source = Vanagon::Component::Source::Http.new(plaintext_url, sum: md5sum, workdir: workdir)
+      http_source = Vanagon::Component::Source::Http.new(plaintext_url, sum: md5sum, workdir: workdir, sum_type: "md5")
       expect(http_source).to receive(:download).and_return(plaintext_filename)
       http_source.fetch
       expect(http_source.dirname).to eq(plaintext_dirname)
     end
   end
+
+  describe "verify" do
+    it "fails with a bad sum_type" do
+      http_source = Vanagon::Component::Source::Http.new(plaintext_url, sum: md5sum, workdir: workdir, sum_type: "md4")
+      expect(http_source).to receive(:download).and_return(plaintext_filename)
+      http_source.fetch
+      expect{http_source.verify}.to raise_error(RuntimeError, /Don't know how to produce a sum/)
```

Comment:
More information **is** often good, but for a test it means you've tightly coupled the expected error value when all you really care about is that an error (or at least the correct error) is raised. The message for the user isn't what's being tested; correct behavior is.